onnxruntime/test/providers/cpu/tensor/gather_op_test.cc

```diff
@@ -76,7 +77,9 @@ TEST(GatherOpTest, Gather_invalid_index_cpu) {
   test.AddInput<int32_t>("indices", {3}, {0LL, 1L, 1000L});
   test.AddOutput<float>("output", {1}, {1.0f});
 
-  test.Run(OpTester::ExpectResult::kExpectFailure, "Mismatch between number of source and target dimensions.");
+  std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+  execution_providers.push_back(DefaultCpuExecutionProvider());
+  test.Run(OpTester::ExpectResult::kExpectFailure, "indices element out of data bounds", {}, nullptr, &execution_providers);
```

Comment:
this is great!

Reply:
as I mentioned above, I'll add another test to exclude cpu.